### PR TITLE
[SPARK-50309][DOCS] Document `SQL Pipe` Syntax

### PR DESCRIPTION
**Pull request by dtenedor**
### What changes were proposed in this pull request?

This PR adds documentation for SQL pipe syntax.

### Why are the changes needed?

It provides a reference table of available operators and describes how the syntax works in each of the supported circumstances.

### Does this PR introduce _any_ user-facing change?

No, this is a documentation-only change.

### How was this patch tested?

N/A

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by gengliangwang** [comment]
@dtenedor The doc looks good to me overall.
Do we consider showing more examples like https://github.com/google/zetasql/blob/master/docs/pipe-syntax.md?

File: docs/sql-pipe-syntax.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+following operators.
+
+#### Aggregation
+
+Aggregation takes place differently using SQL pipe syntax as opposed to regular Spark SQL.
+
+* To perform full-table aggregation, use the `AGGREGATE` operator with a list of aggregate
+expressions to evaluate.<br>
+  This returns one single row in the output table.
+* To perform aggregation with grouping, use the `AGGREGATE` oeprator with a `GROUP BY` clause.<br>
```
Comment (gengliangwang) [inline docs/sql-pipe-syntax.md:100]:
```suggestion
* To perform aggregation with grouping, use the `AGGREGATE` operator with a `GROUP BY` clause.<br>
```

File: docs/sql-pipe-syntax.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+| `AGGREGATE <agg_expr> [[AS] alias], ...`                                                  | Performs full-table aggregation, returning one result row with<br/>a column for each aggregate expression.                                                                                                                            |
+| `AGGREGATE [<agg_expr> [[AS] alias], ...]`<br/>`GROUP BY <grouping_expr> [AS alias], ...` | Performs aggregation with grouping, returning one row per group.<br/>The column list includes the grouping columns first and then the<br/>aggregate columns afterwards. Aliases can be assigned directly<br/>on grouping expressions. |
+| `[LEFT \| ...] JOIN <relation>`<br/>` [ON <condition> \| USING(col, ...)]`                | Joins rows from both inputs, returning a filtered cross-product of<br/>the pipe input table and the table expression following the<br/>JOIN keyword.                                                                                  |
```
Comment (gengliangwang) [inline docs/sql-pipe-syntax.md:157]: Let's put down all the supported join types?

File: docs/sql-pipe-syntax.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+| `ORDER BY <expr> [ASC \| DESC], ...`                                                      | Returns the input rows after sorting as indicated.                                                                                                                                                                                    |
```
Comment (gengliangwang) [inline docs/sql-pipe-syntax.md:158]: Here is the doc for ORDER BY: https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-orderby.html
Do we support nulls_sort_order?

File: docs/sql-pipe-syntax.md (hunk 1 above)
Comment (dtenedor) [reply docs/sql-pipe-syntax.md]: Fixed.

File: docs/sql-pipe-syntax.md (hunk 2 above)
Comment (dtenedor) [reply docs/sql-pipe-syntax.md]: Sounds good, updated.

File: docs/sql-pipe-syntax.md (hunk 3 above)
Comment (dtenedor) [reply docs/sql-pipe-syntax.md]: Yes, we do support it. I added a mention for this.

**Comment by dtenedor** [comment]
@gengliangwang I updated the docs with more examples and information per recommendation, please take another look.

File: docs/sql-pipe-syntax.md [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+|col|
++---+
+|  1|
++---+
+```
+
+#### LIMIT
+
+```sql
+|> LIMIT <n> [OFFSET <m>]
```
[the remark this answers is not in the document]
Comment (cloud-fan) [reply docs/sql-pipe-syntax.md:336]: and `OFFSET` only?

**Review by cloud-fan — APPROVED** [review]
nice doc!

**Review by gengliangwang — APPROVED** [review]
Nice

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by MaxGekk — APPROVED** [review]

File: docs/sql-pipe-syntax.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dtenedor) [reply docs/sql-pipe-syntax.md]: I checked, and `OFFSET` is allowed without `LIMIT`, and vice versa. But if both are present, `LIMIT` must come before `OFFSET`. Updated this part accordingly.

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**